pkg/controller/baremetalasset/baremetalasset_controller.go

```diff
@@ -3,6 +3,7 @@ package baremetalasset
 import (
 	"context"
 	"encoding/json"
+	customError "errors"
```

Comment:
```suggestion
	"errors"
```

Seeing `customError` made me think you were doing something new. Just let it be `"errors"`

Reply:
Ahh, I see there is already another errors package being imported.

Reply:
In that case, change the name of the other one to `apierrors`. It's surprising to see pkgs from core renamed this way.

Reply:
Fixed as suggested.